tests/contrib/webapp2/test_webapp2.py

```diff
@@ -0,0 +1,153 @@
+import json
+
+import webapp2
+
+from ddtrace.constants import SAMPLING_PRIORITY_KEY
+from ddtrace.contrib.webapp2 import Webapp2TraceMiddleware
+from ddtrace.ext import errors, http
+from ...base import BaseTracerTestCase
+
+
+class MyHandler(webapp2.RequestHandler):
+    def post(self):
+        json_body = self.request.json
+        payload = json.dumps(json_body)
+        self.response.write(payload)
+
+
+class Webapp2TestCase(BaseTracerTestCase):
+    def setUp(self):
+        super(Webapp2TestCase, self).setUp()
+        handler = MyHandler
+        route = webapp2.Route('/tests/', handler, methods=['POST'])
+        route1 = webapp2.Route('/unused/', handler, methods=['POST'])
+        routes = [route, route1]
+        self.app = webapp2.WSGIApplication(routes=routes)
+        self.traced_app = Webapp2TraceMiddleware(
+            self.app, self.tracer, service='test-service')
+
+    def test_basic_post(self):
+        # Given a webapp2 app with a simple POST endpoint
+        # When doing a valid POST request
+        data = {'valid': 'test'}
+        payload = json.dumps(data)
+        request = webapp2.Request.blank('/tests/', POST=payload)
+        response = request.get_response(self.app)
+
+        # Then the response is successful
+        assert response.status_code == 200
+        assert response.text == payload
+
+    def test_apm_middleware_is_transparent(self):
+        # Given an application instrumented with our tracing middleware
+        # When doing a basic POST request
+        data = {'valid': 'test'}
+        payload = json.dumps(data)
+        request = webapp2.Request.blank('/tests/', POST=payload)
+        response = request.get_response(self.traced_app)
+
+        # Then the response will be the same as without the middleware
+        assert response.status_code == 200
+        assert response.text == payload
+
+    def test_apm_middleware_valid_post(self):
+        # Given an application instrumented with a tracer which captures spans
+        # When doing a basic POST request
+        data = {'valid': 'test'}
+        payload = json.dumps(data)
+        request = webapp2.Request.blank('/tests/', POST=payload)
```

Comment:
Can you add a test with a query string to see if it's part or not of the final URL?

Reply:
Yup, done. Thanks.